### ISSUE TITLE
Question regarding using links without allowing them

### ISSUE THREAD

**Issue by bobiblazeski** [rexxars/react-markdown#8]
I want to render tags as link to my site #github => [#github]('/tag/github') but I don't want to allow user to use the Link markdown syntax for anything else.
Currently I'm doing this which works but allows user to use the Link too

```
 (<ReactMarkdown source={prepare(this.props.text+' ')}
        softBreak="br"
        allowedTypes={['Text', 'Paragraph', 'Hardbreak', 'Emph', 'Strong', 'Item', 'List','Link']} />

function prepare(text){
  return U.tagify(text).reduce(function(acc, token){
      return acc + (U.isTag(token) ? makeLink(token) : token);
    }.bind(this))
}

function makeLink(token){
  return '['+token+'](/tag/'+token.substr(1)+')';
}

```
 Is there any  way to do that?

**Comment by rexxars** [comment]
I suppose the easiest solution would be to just regex for any non-internal links and remove them (or throw an error if any are present):
```js
mdSource = mdSource.replace(/\[[^\]]+\]\(https?:\/\/.*?\)/g, '');
```

But I've been working on a branch that allows you to pass a function `allowNode` as a prop, which can be used to filter nodes. It should be merged within a few days. The API should be something like this:

```js
React.render(
    <ReactMarkdown
        source={mdSource}
        allowNode={node => node.type !== 'Link' || node.props.href.match(/^\//)}>,
    document.getElementById('md-output')
);
```

It can be paired with the existing `allowedTypes`/`disallowedTypes` for greater flexibility.

**Comment by bobiblazeski** [comment]
Sounds good will check it in a few days.

**Comment by rexxars** [comment]
Just released 1.2.0 which includes the `allowNode` feature.
